### PR TITLE
[CARBONDATA-3564] optimize java code checkstyle for EmptyStatement rule

### PR DESCRIPTION
**Pull request by lamber-ken**
Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [x] Any interfaces changed? NO
 
 - [x] Any backward compatibility impacted? NO
 
 - [x] Document update required? NO

 - [x] Testing done YES
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [x] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.  NO

Detects empty statements (standalone {@code ";"} semicolon).

Empty statements often introduce bugs that are hard to spot

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.1/721/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.2/728/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/729/

**Comment by jackylk** [comment]
LGTM

File: core/src/main/java/org/apache/carbondata/core/metadata/SegmentFileStore.java [hunk 1]
```diff
@@ -491,7 +491,7 @@ public static boolean updateSegmentFile(CarbonTable carbonTable, String segmentI
         LOGGER.error(
             "Not able to acquire the lock for Table status updation for table path " + tablePath);
       }
-      ;
+
```
Comment (akashrn5) [inline core/src/main/java/org/apache/carbondata/core/metadata/SegmentFileStore.java]: this new line is not required, please remove

**Comment by akashrn5** [comment]
@lamber-ken , since this PR also related to checkstyle, can you please map CARBONDATA-3563 to this PR and close CARBONDATA-3564 by mentioning in comment that it is mapped to other jira.

**Pull request closed without merging**